evennia/contrib/turnbattle/tb_range.py

```diff
@@ -0,0 +1,1356 @@
+    # Lastly, move closer to your target.
+    distance_dec(mover, target)
+    return 
+
+def withdraw(mover, target):
+    """
+    Manages a character's whole withdrawal, including changes in ranges to other characters.
+    
+    Args:
+        mover (obj): The object moving
+        target (obj): The object to be moved away from
+        
+    Notes:
+        The mover will also automatically move away from objects that are close to the target
+        of their withdrawl. The mover will never inadvertently move toward anything else while
+        withdrawing - they can be considered to be moving to open space.
+    """
+    objects = mover.location.contents
+    for thing in objects:
+        if thing != mover and thing != target:
+            # Move away from each object closer to the target than you, if it's also closer to you than you are to the target.
+            if mover.db.Combat_Range[thing] >= target.db.Combat_Range[thing] and mover.db.Combat_Range[thing] < mover.db.Combat_Range[thing]:
+                distance_inc(mover, thing)
+            # Move away from anything your target is engaged with
+            if target.db.Combat_Range[thing] == 0:
+                distance_inc(mover, thing)
+            # Move away from anything you're engaged with.
+            if mover.db.Combat_Range[thing] == 0:
+                distance_inc(mover, thing)
+    # Then, move away from your target.
+    distance_inc(mover, target)
+    return
+    
+def get_range(obj1, obj2):
+    """
+    Gets the combat range between two objects.
+    
+    Args:
+        obj1 (obj): First object
+        obj2 (obj): Second object
+        
+    Returns:
+        range (int or None): Distance between two objects or None if not applicable
+    """
+    # Return None if not applicable.
+    if not obj1.db.Combat_Range:
+        return None
+    if not obj2.db.Combat_Range:
+        return None
+    if obj1 not in obj2.db.Combat_Range:
+        return None
+    if obj2 not in obj1.db.Combat_Range:
+        return None
+    # Return the range between the two objects.
+    return obj1.db.Combat_Range[obj2]
+
+def combat_cleanup(character):
+    """
+    Cleans up all the temporary combat-related attributes on a character.
+
+    Args:
+        character (obj): Character to have their combat attributes removed
+
+    Notes:
+        Any attribute whose key begins with 'combat_' is temporary and no
+        longer needed once a fight ends.
+    """
+    for attr in character.attributes.all():
+        if attr.key[:7] == "combat_":  # If the attribute name starts with 'combat_'...
+            character.attributes.remove(key=attr.key)  # ...then delete it!
+
+
+def is_in_combat(character):
+    """
+    Returns true if the given character is in combat.
+
+    Args:
+        character (obj): Character to determine if is in combat or not
+
+    Returns:
+        (bool): True if in combat or False if not in combat
+    """
+    if character.db.Combat_TurnHandler:
```

Comment:
This can simply and more succinctly be expressed as `return bool(character.db.Combat_TurnHandler)`.